packages/xo-web/src/common/render-xo-item.js

```diff
@@ -17,6 +20,108 @@ const OBJECT_TYPE_TO_ICON = {
   network: 'network',
 }
 
+const XoItem = ({ children, item, link, to }) =>
+  item !== undefined ? (
+    link ? (
+      <Link to={to} target='_blank'>
+        {children()}
+      </Link>
+    ) : (
+      children()
+    )
+  ) : (
+    <span className='text-muted'>{_('errorNoSuchItem')}</span>
+  )
+
+XoItem.propTypes = {
+  item: propTypes.object,
+  link: propTypes.bool,
+  to: propTypes.oneOfType([propTypes.string, propTypes.object]),
+}
+// ===================================================================
+
+const XO_ITEM_PROP_TYPES = {
+  id: PropTypes.string.isRequired,
+  link: PropTypes.bool,
+}
+
+export const VmItem = [
+  connectStore(() => {
+    const getVm = createGetObject()
+    return {
+      vm: getVm,
+      container: createGetObject(
+        createSelector(getVm, vm => get(() => vm.$container))
+      ),
+    }
+  }),
+  ({ vm, container, ...props }) => (
+    <XoItem item={vm} to={`/vms/${vm.id}`} {...props}>
+      {() => (
+        <span>
+          <Icon icon={`vm-${vm.power_state.toLowerCase()}`} />{' '}
+          {vm.name_label || vm.id}
+          {container !== undefined &&
+            ` (${container.name_label || container.id})`}
+        </span>
+      )}
+    </XoItem>
+  ),
+].reduceRight((value, decorator) => decorator(value))
+
+VmItem.propTypes = XO_ITEM_PROP_TYPES
+
+export const SrItem = [
+  connectStore(() => {
+    const getSr = createGetObject()
+    return {
+      sr: getSr,
+      container: createGetObject(
+        createSelector(getSr, sr => get(() => sr.$container))
+      ),
+    }
+  }),
+  ({ sr, container, ...props }) => (
+    <XoItem item={sr} to={`/srs/${sr.id}`} {...props}>
+      {() => (
+        <span>
+          <Icon icon='sr' /> {sr.name_label || sr.id}
+          {container !== undefined && (
+            <span className='text-muted'> - {container.name_label}</span>
+          )}
+          {isSrWritable(sr) && (
+            <span>{` (${formatSize(sr.size - sr.physical_usage)} free)`}</span>
+          )}
+        </span>
+      )}
+    </XoItem>
+  ),
+].reduceRight((value, decorator) => decorator(value))
+
+SrItem.propTypes = XO_ITEM_PROP_TYPES
+
+export const RemoteItem = [
+  addSubscriptions(({ id }) => ({
+    remote: cb =>
+      subscribeRemotes(remotes => {
+        cb(get(() => remotes.find(remote => remote.id === id)))
```

Comment:
Is `get` necessary here?

Reply:
remotes can be undefined when it's not fetched yet.